backend/src/journey/controller/journey.controller.ts

```diff
@@ -0,0 +1,28 @@
+import { Body, Controller, Post } from "@nestjs/common";
+import { ApiBadRequestResponse, ApiOkResponse, ApiOperation, ApiTags } from "@nestjs/swagger";
+import { BadRequest } from "app/entity/badrequest.entity";
+import { JourneyRequestDto } from "journey/dto/journeyRequest.dto";
+import { Journey } from "journey/entity/journey.entity";
+import { JourneyService } from "../service/journey.service";
+
+@Controller("journey")
+@ApiTags("journey")
+export class JourneyController {
+  constructor(private readonly journeyService: JourneyService) { }
+
+  @Post()
+  @ApiOperation({
+    summary: "queries a journey"
+  })
+  @ApiOkResponse({
+    description: "Journeys.",
+    type: [Journey]
+  })
+  @ApiBadRequestResponse({
+    description: "Bad request.",
+    type: BadRequest
+  })
+  public queryJourney(@Body() requestBody: JourneyRequestDto): Promise<Journey[]> {
+    return this.journeyService.queryJourney(requestBody.originId, requestBody.destinationId/*, requestBody.departure*/);
```

Comment:
The departure time is missing here. Any reason for that?

Reply:
Departure time and "asArrival" is passed to the service now. But vrr_client is not supporting those two parameters yet. Should we create a new issue for that?

Reply:
Okay I see know. Maybe the service should ignore the parameter than and I will create an issue to implement that.

Reply:
267